lib/manifestly/entity/base.rb

```diff
@@ -0,0 +1,60 @@
+module Manifestly
+  module Entity
+    class Base
+      @attributes = nil
+
+      def initialize(data = {})
+        invalids = data.keys - attributes
+        raise "The following invalid #{self.class} keys were found: #{invalids}" unless invalids.empty?
+
+        self.attributes = data
+      end
+
+      def attributes
+        self.class.attributes
+      end
+
+      def self.attributes
+        @attributes ||= []
+      end
+
+      def attributes=(attrs)
+        attrs.each { |k, v| send(:"#{k}=", v) }
+      end
+
+      def self.attr_accessor(*attrs)
+        attrs.each { |attr| attributes << attr }
+        super
```

Comment:
I am not sure how I feel about this.  Its probably fine, but we can no longer have attr_accessors that are not part of the @attributes array we are building.  We likely won't have much need for "normal" attr_accessors (and use normal methods if we need that functionality), but adding an attr_accessor and getting the magic functionality here is a little surprising if you aren't expecting it.  Could rename this method, or add a code comment by the attr_xxxxors list in each file, or leave it as is. /shrug